topics/RSA-Keys-Generation.md

```diff
@@ -0,0 +1,240 @@
+* `tee ktor.spki`: The `tee` command is used to redirect the output to both the console and a file named `ktor.spki`. This file will contain the public key in PEM format.
+</tab>
+</tabs>
+
+With the public key at hand, you can now derive its exponent and modulus values.
+
+## Extracting the modulus & exponent attributes
+
+Now that you have your key-pair, you need to extract the `e` (exponent) and `n` (modulus) attributes of your
+public key, in order to use them in your `jwks.json` file. To do this using OpenSSL, you need to:
+
+1) Read the public key from the `.spki` file you created
+2) Display information about the key in a human-readable format
+
+This can easily be done with the following command:
+
+<tabs>
+<tab title="OpenSSL">
+<code-block lang="shell">
+openssl pkey -in ktor.spki -pubin -noout -text
+</code-block>
+
+* `pkey`: This is the OpenSSL command-line utility for processing private and public keys.
+* `-in ktor.spki`: Specifies the input file containing the public key in PEM format. In this case, the input file is ktor.spki.
+* `-pubin`: Indicates that the input file contains a public key. Without this option, OpenSSL would assume that the input file contains a private key.
+* `-noout`: This option prevents OpenSSL from outputting the encoded public key. The command will only display information about the public key, and the actual key won't be printed to the console.
+* `-text`: Requests that OpenSSL display the textual representation of the key. This includes details such as the key type, size, and the actual key data in human-readable form.
+</tab>
+</tabs>
+
+Let's see the sample output by using the key present in the official ktor [auth-jwt-rs256 sample](https://github.com/ktorio/ktor-documentation/tree/1.6.8/codeSnippets/snippets/auth-jwt-rs256).
+You can find the private key in the [applciation.conf](https://github.com/ktorio/ktor-documentation/blob/1.6.8/codeSnippets/snippets/auth-jwt-rs256/src/main/resources/application.conf) file.
+
+```Shell
+$ openssl pkey -in ktor.spki -pubin -noout -text
+RSA Public-Key: (512 bit)
+Modulus:
+    00:b5:f2:5a:2e:bc:d7:20:b5:20:d5:4d:cd:d4:a5:
+    7c:c8:9a:fd:d8:61:e7:e4:eb:58:65:1e:ea:5a:4d:
+    4c:73:87:32:e0:91:a3:92:56:2e:a7:bc:1e:32:30:
+    43:f5:fd:db:05:5a:08:b2:25:15:5f:ac:4d:71:82:
+    2b:d0:87:b4:01
+Exponent: 65537 (0x10001)
+```
+<warning>
+<p>
+The sample key uses 512-bit, which is not secure. We are using that only as an example! Ideally, you should
+opt for 2048-bit or 4096-bit keys.
+</p>
+</warning>
+
+You now have the `n` and `e` attributes needed for your `jwks.json` file, but they are in hexadecimal format. Let's
+see how to convert them to decimal and encode them properly.
+
+##  Converting and encoding the modulus & exponent attributes
+
+We now need to convert the hexadecimal representation of the exponent and modulus to their respective [Base64URL](https://en.wikipedia.org/wiki/Base64#URL_applications) encodings.
```

Comment:
There's a mixed use of "You" and "We". I suggest you pick one, probably the one used in the other `.md` files, for consistency.

Reply:
Good catch! Should be fixed along with references to "our".